domains_plugin.php

```diff
@@ -150,14 +150,45 @@ public function install($plugin_id)
      */
     public function upgrade($current_version, $plugin_id)
     {
+        Loader::loadModels($this, ['Actions', 'Companies', 'Navigation']);
+        Configure::load('domains', dirname(__FILE__) . DS . 'config' . DS);
         if (!isset($this->Record)) {
             Loader::loadComponents($this, ['Record']);
         }
 
         // Upgrade if possible
         if (version_compare($this->getVersion(), $current_version, '>')) {
-            // Upgrade to v1.1.0
+            // Handle the upgrade, set errors using $this->Input->setErrors() if any errors encountered
+
+            // Upgrade to 1.1.0
             if (version_compare($current_version, '1.1.0', '<')) {
+                $companies = $this->Companies->getAll();
+                foreach ($companies as $company) {
+                    // Get the current Packages > Domains action
+                    $action = $this->Actions->getByUrl(
+                        'plugin/domains/admin_domains/browse/',
+                        'nav_staff',
+                        $company->id
+                    );
+
+                    // Gets the nav item for Billing > Services
+                    $services_nav = $this->Navigation->getAll(
+                        ['url' => 'billing/services/', 'location' => 'nav_staff', 'company_id' => $company->id]
+                    );
+
+                    if ($action) {
```

Comment:
Looking at the code I can see that you are trying to add the "Domains" link below "Services", but this seems to be unreliable, as I got 2 different results (and none of them was the expected one). 

Upgrading from 1.0.0 to 1.1.0:
![Screen Shot 2021-06-16 at 1 31 58 PM](https://user-images.githubusercontent.com/23648083/122281196-3ccf2200-cea7-11eb-9f92-98a00a08bce5.png)

On a fresh installation:
![Screen Shot 2021-06-16 at 1 33 37 PM](https://user-images.githubusercontent.com/23648083/122281411-77d15580-cea7-11eb-8eb3-e986f016c274.png)

Reply:
Well I fixed the upgrade, but I don't know there's anything I can do about the install since the actions are added after install() is executed.

Reply:
A possible solution is to implement a new event for when a plugin is installed, similar to "ModuleManager.add", but I think it is a very complex task for something that simple. 🤔

Reply:
Yea, I'm thinking that inserting a nav item above/below another should be supported by the core somehow.